### ISSUE TITLE
Move away from PhantomJS as default?

### ISSUE THREAD

**Issue by rmm5t** [jejacks0n/teaspoon#545]
With the [PhantomJS project abandoned](https://github.com/ariya/phantomjs/issues/15344), and [potential issues with teaspoon + phantomjs](https://github.com/jejacks0n/teaspoon/issues/527#issuecomment-419500403), should teaspoon change it's default driver to something else?

**Comment by jejacks0n** [comment]
Yes, for sure.

**Comment by jejacks0n** [comment]
I think we should just build out the selenium setup, specifically focusing on chromedriver.

**Comment by thbar** [comment]
I second that - on a client's project, the build started failing on CircleCI, I _think_ because PhantomJS was moved to "legacy" (https://circleci.com/docs/2.0/circleci-images/#language-image-variants). I suspect we'll see more of that in the future on other cases.

**Comment by teeparham** [comment]
For reference, to switch to selenium + chromedriver:

```rb
# Gemfile
group :test do
  gem "selenium-webdriver"
end

# teaspoon_env.rb
Teaspoon.configure do |config|
  config.driver = :selenium
  config.driver_options = { client_driver: :chrome }
end
```

**Comment by mathias** [comment]
I seem to be caught in a Catch-22 in using Chrome with selenium-webdriver, though.

If I have the gem `gem "chromedriver"` in my Gemfile, then I can run my Capybara acceptance tests in Rspec. But Teaspoon doesn't like it.

```
Selenium::WebDriver::Error::WebDriverError: unable to connect to chromedriver 127.0.0.1:9515
  /Users/mgauger/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/selenium-webdriver-3.14.1/lib/selenium/webdriver/common/service.rb:142:in `connect_until_stable'
```

If I remove the `gem "chromedriver"` in the Gemfile, it finds the system Chromedriver (installed with Brew on this mac) and runs the Teaspoon suite. But Capybara tests fail with errors like:

```  1) user views the page
     Failure/Error: visit '/page'

     Selenium::WebDriver::Error::WebDriverError:
       unable to connect to chromedriver 127.0.0.1:9515
```

Has anyone else run into this?

**Comment by joshm1204** [comment]
Has anyone looked into headless browsers?  I reference this article and would love feedback on if this is a good direction to go. I am new to Teaspoon, so not sure.

https://www.puzzle.ch/de/blog/articles/2018/02/12/phantomjs-is-dead-long-live-headless-browsers

**Comment by rmm5t** [comment]
@joshm1204 I think the suggested default approach here would be "Alternative 2" in the article you referenced.

**Comment by joshm1204** [comment]
Yes, I think a chrome's headless browser for example would be a good solution. Is there any ability to use Chrome's headless browser yet?

**Comment by rmm5t** [comment]
@joshm1204 In addition to either using chromedriver-helper gem (deprecated) or webdrivers gem (replacement) and configuring the chromedriver version, you can do something like this inside the `teaspoon_env.rb`:

```ruby
  config.driver = :selenium

  options = ::Selenium::WebDriver::Chrome::Options.new(args: ["--no-sandbox"])
  options.headless!
  config.driver_options = { client_driver: :chrome, client_driver_opts: { options: options } }
```

**Comment by la-ruby** [comment]
i'm using a modified version of ^ snippet to, like solidus does at
`solidus/backend/spec/teaspoon_env.rb`.
otherwise a chrome window appears , which shouldn't if it was truly headless.
```
  config.driver = :selenium
  config.driver_options = {
    client_driver: :chrome,
    selenium_options: {
      options: Selenium::WebDriver::Chrome::Options.new(
        args: %w(headless disable-gpu window-size=1920,1440),
      ),
    },
  }
```
